node/src/citadel.js

```diff
@@ -0,0 +1,97 @@
+import fetch from "node-fetch"
+
+const { API_CLIENT_ID, API_SECRET, API_URL } = process.env
+
+/**
+ * Returns the default headers used when interacting with Citadel
+ * @return The headers used when making requests to Citadel
+ **/
+const getHeaders = () => {
+  return {
+    "Content-Type": "application/json",
+    Accept: "application/json",
+    "X-Access-Client-Id": API_CLIENT_ID,
+    "X-Access-Secret": API_SECRET,
+  }
+}
+
+/**
+ * Calls out to Citadel exchanging the public token given by the API request
+ * for an access token to make subsequent requests
+ * @param {String} public_token The token provided by the API request to exchange
+ * @return The access token provided by citadel
+ **/
+const getAccessToken = async (public_token) => {
```

Comment:
calls to citadel API look duplicated, 
it will be better to remove duplication

Reply:
duplicated in what sense?

Reply:
i mean  you can use generic request for different endpoints

```
function request(url, data, errorText) {
  try {
    const response = await fetch(url, {
      method: "POST",
      body: inputBody,
      headers: headers,
    })
    const body = await response.json()
    return body
  } catch (e) {
    console.error(errorText)
    console.error(e)
    throw e
  }

}
```

Reply:
refactored per your suggestion